cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2032,17 +2047,32 @@ private Map<ProgramRunId, RunRecordDetail> getAllProgramRuns(ProgramReference pr
       String recordType) throws IOException {
     List<Field<?>> prefix = getRunRecordProgramRefPrefix(recordType, programReference);
     Range scanRange = createRunRecordScanRange(prefix, startTime, endTime);
-    return getRuns(scanRange, status, limit, null, filter);
+    // Need to specify that the result is ordered by run_start_time since
+    // the version field is not provided
```

Comment:
nit: Can we add some additional details to the comment on what goes wrong when version field is not provided, thanks!